### PR TITLE
Update App Configuration management plane API description

### PR DESCRIPTION
**Pull request by jimmyca15**
Update App Configuration management plane API description to declare intent behind key-value management plane API.

### CODE REVIEW

**Review by jiayi11 — APPROVED** [review]

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json [hunk 1]
```diff
@@ -821,7 +821,7 @@
         "tags": [
           "KeyValues"
         ],
-        "description": "Gets the properties of the specified key-value.",
+        "description": "Gets the properties of the specified key-value. NOTE: This operation is intended for use in ARM Template deployments. The data plane API should be used for other interactions with App Configuration key-values.",
```
Comment (zhenlan) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json:824]: Not sure if people will interpret "other interactions" as "other operations but this one". I know you mean "other scenarios but the ARM template".

Do you feel if it makes sense to include a link to our data-plane REST API doc?
https://go.microsoft.com/fwlink/?linkid=2078296

Comment (jimmyca15) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json:824]: > Do you feel if it makes sense to include a link to our data-plane REST API doc?

I was a bit on the fence about it. Wasn't sure if it made sense to link to the dataplane REST API spec (in this repo), the link you linked, or SDK link which ultimately this is used to build.

I'm fine to add the link if you'd like.

Comment (zhenlan) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json:824]: > NOTE: This operation is intended for use in ARM Template deployments. For all other scenarios involving App Configuration key-values the data plane API should be used instead.

Love it.

> I was a bit on the fence about it. 

I was just throwing an idea. If a link looks like an outlier to you, I'm okay not having it.

Comment (jimmyca15) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json]: I'll omit the link then.

Updated the wording, thanks !

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json [hunk 2]
```diff
@@ -821,7 +821,7 @@
         "tags": [
           "KeyValues"
         ],
-        "description": "Gets the properties of the specified key-value.",
+        "description": "Gets the properties of the specified key-value. NOTE: This operation is intended for use in ARM Template deployments. For all other scenarios involving App Configuration key-values the data plane API should be used instead.",
```
Comment (zhenlan) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json:824]: Just want to get your thoughts... How do you feel about mentioning Bicep/Terraform besides of ARM template?

Comment (jimmyca15) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json:824]: my thought is to stick with the most fundamental piece. Kind of how SDK team refers to their JavaScript (and TypeScript) SDK as the JS SDK.

Comment (zhenlan) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json:824]: Sounds good.

**Review by zhenlan — APPROVED** [review]

**Comment by jimmyca15** [comment]
#sign-off

**Review by v-jiaodi — APPROVED** [review]

**Review by qiaozha — APPROVED** [review]

**Pull request merged**